packages/fluentui/react-bindings/test/styles/resolveVariables-test.ts

```diff
@@ -1,84 +1,120 @@
+    expect(variables).toMatchObject({ backgroundColor: 'blue' });
   });
 
-  test('considers displayName while caching resolved variables', () => {
-    const componentVariablesMock1 = jest.fn().mockReturnValue({ backgroundColor: 'blue' });
-    const componentVariablesMock2 = jest.fn().mockReturnValue({ color: 'red' });
-    const theme = createTheme();
-    theme.componentVariables['Test1'] = componentVariablesMock1;
-    theme.componentVariables['Test2'] = componentVariablesMock2;
-
-    const variables = resolveVariables('Test1', theme, {}, true);
-    const secondVariables = resolveVariables('Test2', theme, {}, true);
+  test('avoid merging for multiple display names if variables are undefined', () => {
+    const theme = createTheme({ Foo: testVariables, Bar: testVariables });
+    const variables = resolveVariables(['Foo', 'Bar'], theme, undefined, false);
 
     expect(variables).toMatchObject({ backgroundColor: 'blue' });
-    expect(secondVariables).toMatchObject({ color: 'red' });
-    expect(componentVariablesMock1).toHaveBeenCalledTimes(1);
-    expect(componentVariablesMock2).toHaveBeenCalledTimes(1);
   });
 
-  test('considers theme while caching resolved variables', () => {
-    const componentVariablesMock = jest.fn().mockReturnValue({ backgroundColor: 'blue' });
-    const theme = createTheme('Test');
-    const secondTheme = createTheme('Test');
-    theme.componentVariables['Test'] = componentVariablesMock;
-    secondTheme.componentVariables['Test'] = componentVariablesMock;
+  describe('enabledVariablesCaching', () => {
+    test('caches resolved variables', () => {
+      const fooVariables = jest.fn().mockReturnValue({ backgroundColor: 'blue' });
+      const theme = createTheme({ Foo: fooVariables });
 
-    const variables = resolveVariables('Test', theme, {}, true);
-    const secondVariables = resolveVariables('Test', secondTheme, {}, true);
+      const variables = resolveVariables(['Foo'], theme, {}, true);
+      const secondVariables = resolveVariables(['Foo'], theme, {}, true);
 
-    expect(variables).toMatchObject({ backgroundColor: 'blue' });
-    expect(secondVariables).toMatchObject({ backgroundColor: 'blue' });
-    expect(componentVariablesMock).toHaveBeenCalledTimes(2);
+      expect(variables).toMatchObject(secondVariables);
+      expect(fooVariables).toHaveBeenCalledTimes(1);
+    });
+
+    test('omits usage of undefined variables', () => {
+      const fooVariables = jest.fn().mockReturnValue({ content: 'foo' });
+      const theme = createTheme({ Foo: fooVariables, Bar: undefined });
+
+      expect(resolveVariables(['Foo', 'Bar'], theme, {}, true)).toMatchObject({ content: 'foo' });
+      expect(fooVariables).toHaveBeenCalledTimes(1);
+    });
+
+    test('handles multiple displayNames', () => {
+      const fooVariables = jest.fn().mockReturnValue({ backgroundColor: 'blue' });
+      const barVariables = jest.fn().mockReturnValue({ color: 'red' });
```

Comment:
Would suggest adding a scenario with ovedrriding variabels too.

Reply:
I changed this test to handle it:

```tsx
const fooVariables = jest
  .fn()
  .mockReturnValue({ backgroundColor: "blue", borderColor: "black" });
const barVariables = jest
  .fn()
  .mockReturnValue({ backgroundColor: "green", color: "red" });
const theme = createTheme({ Foo: fooVariables, Bar: barVariables });

expect(resolveVariables(["Foo", "Bar"], theme, {}, true)).toMatchObject({
  backgroundColor: "blue",
  borderColor: "black",
  color: "red"
});
```